### PR TITLE
Add dockerfile and test so we can test simple remoting behaviors

### PR DESCRIPTION
**Pull request by JamesWTruher**
this docker file creates an image which can be used to test remoting with containers.
We start with the microsoft/windowsservercore image and then set up the system for
testing by adding a user, setting up configuration for basic auth, installing
PowerShell Core, and creating a configuration endpoint for PSCore.

The tests are very simple; it retrieve $PSVersionTable.PSVersion in the following
connections:
   Full -> Full
   Full -> Core
   Core -> Full
   Core -> Core

This is the starting point for our remoting testing using containers

### CODE REVIEW

File: test/docker/networktest/DockerRemoting.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        if ( $dockerimage -ne $imageName ) {
+            $pending = $true
+            write-warning "Docker image '$imageName' not found, not running tests"
+            return
+        }
+        else {
+            $pending = $false
+        }
+        # this is all I could figure out how to set up a container which
+        # sticks around for a while
```
Comment (mirichmo) [inline test/docker/networktest/DockerRemoting.Tests.ps1]: What about the container lifecycle commands? `docker container start`, `docker exec`, `docker container stop`, etc? `docker ps` and `docker container list` both show running containers.

Using those commands should prevent the need to use start-sleep.

File: test/docker/networktest/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# escape=`
+FROM microsoft/windowsservercore
+
+SHELL ["powershell.exe","-command"]
+
+# the source msi should change on a daily basis
+# the destination should not change
+ADD PSCore.msi /PSCore.msi
```
Comment (mirichmo) [inline test/docker/networktest/Dockerfile]: How does this file get to the Dockerfile "root" directory? Shouldn't it be downloaded from Myget?

File: test/docker/networktest/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+# the msipath (PSCore.msi) below will need to change based on the daily package
+# set up for basic auth
+# install-powershellremoting will restart winrm service
+RUN net user testuser 11aa!!AA /add; `
```
Comment (mirichmo) [inline test/docker/networktest/Dockerfile]: As long as you are using PowerShell...
New-LocalUser -Name testuser -Password (ConvertTo-SecureString 11aa!!AA -AsPlainText -Force)

File: test/docker/networktest/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    net localgroup administrators testuser /add; `
```
Comment (mirichmo) [inline test/docker/networktest/Dockerfile]: Add-LocalGroupMember -Group Administrators -Member testuser

File: test/docker/networktest/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    set-item wsman:/localhost/service/auth/basic $true; `
+    set-item WSMan:/localhost/client/trustedhosts "*" -force; `
+    set-item wsman:/localhost/service/AllowUnencrypted $true; `
+    set-item WSMan:/localhost/client/AllowUnencrypted $true; `
+    Start-Process -FilePath msiexec.exe -ArgumentList '-qn', `
+    '-i c:\PSCore.msi','-log c:\PSCore-install.log','-norestart' -wait ; `
+    $psexec = get-item -path ${ENV:ProgramFiles}/powershell/*/powershell.exe; `
+    $corehome = $psexec.directory.fullname; `
```
Comment (mirichmo) [inline test/docker/networktest/Dockerfile]: You should delete the MSI within this layer to save space

**Review by mirichmo — CHANGES_REQUESTED** [review]

File: test/docker/networktest/Dockerfile (hunk 2 above)
Comment (TravisEz13) [inline test/docker/networktest/Dockerfile]: We don't have the whole package in MyGet, but I've given Jim feedback on how to get the latest MSI from AppVeyor.

Comment (JamesWTruher) [inline test/docker/networktest/Dockerfile]: this is done with the New-DockerTestBuild.ps1 script. I broke it out to it's own function.

File: test/docker/networktest/Dockerfile (hunk 5 above)
Comment (JamesWTruher) [inline test/docker/networktest/Dockerfile]: heh - 23mb over a 10GB image - not sure it's a big saving, but ok.

**Review by mirichmo — CHANGES_REQUESTED** [review]

File: test/docker/networktest/New-DockerTestBuild.ps1 [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+    param(
+        [Parameter(Mandatory)]
+        [string]$build,
+        [string]$downloadFolder = '.',
+        [string]$artifactFilter = '*.msi',
+        [string]$artifactLocalFileName = $Constants.MsiName
+    )
+
+    $headers = @{
+      'Authorization' = "Bearer {0}" -f $Constants.Token
```
Comment (JamesWTruher) [inline test/docker/networktest/New-DockerTestBuild.ps1]: fixed

File: test/docker/networktest/New-DockerTestBuild.ps1 [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+}
+
+# Get a collection of appveyor objects representing the builds for the last day
+function Get-AppVeyorBuilds
+{
+    [CmdletBinding()]
+    param(
+        [ValidateNotNullOrEmpty()]
+        [string]$ExtensionBranch = 'master',
+        [string]$Status = "success",
```
Comment (JamesWTruher) [inline test/docker/networktest/New-DockerTestBuild.ps1]: fixed

File: test/docker/networktest/New-DockerTestBuild.ps1 [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+        [ValidateRange(1,7)][int]$Days = 7,
+        [int]$Last = 1
+    )
+
+    [datetime]$start = ((get-date).AddDays(-${Days}))
+    $results = Get-AppVeyorBuildRange -Start $start -LastBuildId $null -ExtensionBranch $ExtensionBranch
+    $results
+    #$results |?{$_.status -match "success"} | sort-object StartedDay | Select -First $Last
```
Comment (JamesWTruher) [inline test/docker/networktest/New-DockerTestBuild.ps1]: fixed

File: test/docker/networktest/New-DockerTestBuild.ps1 [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+
+    if($LastBuildId)
+    {
+        $startBuildIdString = "&startBuildId=$LastBuildId"
+    }
+
+    $URI = "{0}/projects/{1}/{2}/history?recordsNumber={3}{4}&branch{5}" -f $Constants.ApiUrl,$Constants.AccountName,
+        $Constants.ProjectName,$Records,$startBuildIdString,$ExtensionBranch
+    $project = Invoke-RestMethod -Method Get -Uri $URI
+    # "$($Constants.ApiUrl)/projects/$($Constants.AccountName)/$($Constants.ProjectName)/history?recordsNumber=$Records$startBuildIdString&branch=$ExtensionBranch"
```
Comment (JamesWTruher) [inline test/docker/networktest/New-DockerTestBuild.ps1]: fixed

File: test/docker/networktest/New-DockerTestBuild.ps1 [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+
+#### DOCKER OPS #####
+# is docker installed?
+$dockerExe = get-command docker -ea silentlycontinue
+if ( $dockerExe.name -ne "docker.exe" ) {
+    throw "Cannot find docker, is it installed?"
+}
+# Check to see if we already have an image, and if so
+# delete it if -Force was used, otherwise throw and exit
+$TestImage = docker images remotetestimage --format '{{.Repository}}'
```
Comment (JamesWTruher) [inline test/docker/networktest/New-DockerTestBuild.ps1]: oversight - fixed

File: test/docker/networktest/New-DockerTestBuild.ps1 [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+    {
+        docker rmi $Constants.TestImageName
+    }
+    else
+    {
+        throw ("{0} already exists, use '-Force' to remove" -f $Constants.TestImageName)
+    }
+}
+# check again - there could be some permission problems
+$TestImage = docker images remotetestimage --format '{{.Repository}}'
```
Comment (mirichmo) [inline test/docker/networktest/New-DockerTestBuild.ps1]: Here too

**Review by mirichmo — CHANGES_REQUESTED** [review]

**Comment by JamesWTruher** [comment]
i think i have them all

**Review by mirichmo — APPROVED** [review]

**Comment by JamesWTruher** [comment]
thanks!

**Pull request merged**